app/controllers/submissions_controller.rb

```diff
@@ -8,6 +8,28 @@ class SubmissionsController < ApplicationController
   def show; end
 
   def update
+    if params[:submission][:judging_status].present?
+      handle_judging_status_update
+    else
+      handle_comments_update
+    end
+  end
+
+  private
+
+  def handle_judging_status_update
+    unless valid_status_change?
+      return respond_with_error
+    end
+
+    if @submission.update(submission_params)
```

Comment:
as an example of how to write the model validation, you can add the method like this... [more details here](https://guides.rubyonrails.org/active_record_validations.html#performing-custom-validations)
```
class Submission < ApplicationRecord
  validate :can_be_winner, if: &:winner? 
  validate :can_be_ineligible_for_evaluation if: -> { judging_status_change == [:selected, :not_selected] }

  private
  def can_be_winner
    errors.add(:judging_status, "can't be selected to advance when not eligible for evaluation") unless judging_status_was == :selected
    errors.add(:judging_status, "can't be selected to advance until all evaluations are complete") unless all_evaluations_completed?
  end
end
```

```suggestion
    if @submission.update(submission_params)
```